src/v/archival/ntp_archiver_service.cc

```diff
@@ -2016,29 +2018,39 @@ ss::future<> ntp_archiver::garbage_collect_archive() {
     }
 
     size_t successful_deletes{0};
+    size_t successful_segment_deletes{0};
+    size_t segments_in_batch{0};
     const size_t batch_size = 1000;
     std::vector<cloud_storage_clients::object_key> rem_batch;
-    for (const auto& path : segments_to_remove) {
-        vlog(_rtclog.info, "Deleting segment from cloud storage: {}", path);
+    for (const auto& path : objects_to_remove) {
+        std::string_view path_view{path.c_str()};
+        if (!path_view.ends_with("index") && !path_view.ends_with("tx")) {
```

Comment:
Not neccessary in this PR, but at some point helpers for this would be useful: the same condition is used at the top of `cache::trim` (easy to forget if paths ever change)